### PR TITLE
Sync up with master: Part 1

### PR DESCRIPTION
**Pull request by REASY**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1712)

### CODE REVIEW

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
@carloscaldas, @JustinPihony 
What do you think should be proper way to bring those changes back from master to develop? I'm simply cherry-picked those commits from master which are not in develop (my changes) via `git cherry origin/develop origin/master`

**Comment by REASY** [comment]
test!

**Comment by JustinPihony** [comment]
@REASY I think cherry pick is a fine way to go

**Comment by carloscaldas** [comment]
@REASY one thing that I dislike in gitflow is that we cannot protect master branch (due to hotfixes) and cannot make sure a test ran before merging. Since the need happened cherry-pick is ok. But it is better to avoid merging in master unless is a hotfix.

File: src/test/scala/beam/utils/osm/WayFixerTest.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.scalatest.{Matchers, WordSpec}
+
+class WayFixerTest extends WordSpec with Matchers {
+  "A WayFixer" should {
+    "be able to fix lanes" in {
+      {
+        val w = new Way()
+        w.addTag(WayFixer.LANES_TAG, "3");
+        WayFixer.fixLanes(1, w) shouldBe false
+        w.getTag(WayFixer.LANES_TAG) should be("3")
```
Comment (carloscaldas) [inline src/test/scala/beam/utils/osm/WayFixerTest.scala]: extract a method for those duplicate blocks of code

File: src/main/scala/beam/utils/osm/WayFixer.scala [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+//    getFixedSpeed(osmId, way).exists { maxSpeed =>
+//      replace(osmId, way, MAXSPEED_TAG, maxSpeed.toString)
+//      true
+//    }
+//  }
+
+//  private[osm] def getFixedSpeed(osmId: Long, way: Way): Option[Double] = {
+//    Option(way.getTag(MAXSPEED_TAG)) match {
+//      case Some(rawMaxSpeed) =>
+//        if (rawMaxSpeed.startsWith("[")) {
```
Comment (carloscaldas) [inline src/main/scala/beam/utils/osm/WayFixer.scala]: remove commented code

File: src/main/java/beam/router/r5/R5MnetBuilder.java [hunk 3]
```diff
@@ -56,18 +57,21 @@ public R5MnetBuilder(TransportNetwork r5Net, BeamConfig beamConfig) {
...
         // Load the OSM file for retrieving the number of lanes, which is not stored in the R5 network
-//        OSM osm = new OSM(osmFile);
         Map<Long, Way> ways = new OSM(osmFile).ways;
+        WayFixer$.MODULE$.fix(ways);
+
         EdgeStore.Edge cursor = r5Network.streetLayer.edgeStore.getCursor();  // Iterator of edges in R5 network
         OsmToMATSim OTM = new OsmToMATSim(mNetwork, true);
+
+        int numberOfFixes = 0;
+        HashMap<String, Integer> highwayTypeToCounts = new HashMap<>();
```
Comment (carloscaldas) [inline src/main/java/beam/router/r5/R5MnetBuilder.java]:
```Java
Map<String, Integer> highwayTypeToCounts = new HashMap<>();
```

**Comment by REASY** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by JustinPihony** [comment]
The current build is failing because it is failing in master already...per https://github.com/LBNL-UCB-STI/beam/commit/b440293b9de60232128dfa7e761dbd247934234a Many tests are not set up for this injection - https://github.com/LBNL-UCB-STI/beam/blob/master/src/main/scala/beam/agentsim/agents/PersonAgent.scala#L220

**Comment by JustinPihony** [comment]
I'll put together a fuller PR when I get back to this (have to stop for now) - but here is a quick fix I did that works for the OtherPersonAgentSpec:

```
trait InjectableMock extends MockitoSugar {
  val injector  = mock[com.google.inject.Injector](withSettings().stubOnly())
  val config: BeamConfig
  val beamSvc: BeamServices
  private val x = new beam.router.TravelTimeObserved(config, beamSvc)
  when(injector.getInstance(classOf[beam.router.TravelTimeObserved])).thenReturn(x)
}

...

with InjectableMock

...

when(theServices.injector).thenReturn(injector)
```

Note that this is a bit of a circular reference, but solved by laziness...

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Pull request merged**